src/collaboration-platform/GitHubCollaborationPlatform.ts

```diff
@@ -17,25 +41,126 @@ export class GitHubCollaborationPlatform implements CollaborationPlatform {
-    // TODO: Implement
-    return null;
+    const owner = (await repoPathToOwnerAndRepo(this.repoPath)).owner;
+    const repo = (await repoPathToOwnerAndRepo(this.repoPath)).repo;
+    try {
+      const {
+        data,
+      } = await this.octokit.repos.listPullRequestsAssociatedWithCommit({
+        owner,
+        repo,
+        commit_sha: commit.hash,
+      });
+      const prResult = data[0];
+      return {
+        number: prResult.number,
+        url: prResult.url,
+        title: prResult.title,
+        description: prResult.body,
+        isOutdated: prResult.head.sha !== commit.hash,
+      };
+    } catch (e) {
+      return null;
+    }
   }
 
   async getPR(prNumber: PullRequestID): Promise<PullRequestInfo | null> {
-    // TODO: Implement
-    return null;
+    const owner = (await repoPathToOwnerAndRepo(this.repoPath)).owner;
+    const repo = (await repoPathToOwnerAndRepo(this.repoPath)).repo;
+    try {
+      const prResult = this.octokit.pulls.get({
+        owner,
+        repo,
+        pull_number: prNumber,
+      });
+
+      return {
+        number: prNumber,
+        url: (await prResult).data.url,
+        title: (await prResult).data.title,
+        description: (await prResult).data.body,
+      };
+    } catch (e) {
+      return null;
+    }
   }
 
   async createOrUpdatePRForCommits(
-    commitBranchPairs: { commit: Commit; branch: BranchName }[],
+    commitBranchPairs: {
+      commit: Commit;
+      headBranch: BranchName;
+      baseBranch: BranchName;
+    }[],
   ): Promise<Commit[]> {
-    // TODO: Implement
-    return [];
+    let commitArr = new Array();
+    const owner = (await repoPathToOwnerAndRepo(this.repoPath)).owner;
+    const repo = (await repoPathToOwnerAndRepo(this.repoPath)).repo;
+    commitBranchPairs.forEach(async (pair) => {
+      //If the commit has an existing PR, don't create one, but update title + base branch
+      const prResult = await this.getPRForCommit(pair.commit);
+      if (prResult !== null) {
+        await this.octokit.pulls.update({
+          owner,
+          repo,
+          pull_number: prResult.number,
+          title: pair.commit.title,
+          base: pair.baseBranch,
+        });
+      }
+      //Else if the commit doesn't have an existing PR, create one
+      else {
+        await this.octokit.pulls.create({
+          owner,
+          repo,
+          title: pair.commit.title,
+          head: pair.headBranch,
+          base: pair.baseBranch,
+          maintainer_can_modify: true,
+        });
+      }
+      commitArr.push(pair.commit);
+    });
+    return commitArr;
   }
 
   async updatePRDescriptionsForCommitGraph(
     commitPrInfoPairs: { commit?: Commit; prInfo: PullRequestInfo }[],
   ): Promise<void> {
-    // TODO: Implement
+    const owner = (await repoPathToOwnerAndRepo(this.repoPath)).owner;
+    const repo = (await repoPathToOwnerAndRepo(this.repoPath)).repo;
+    // Get list of PRs
```

Comment:
I know this is copied from the old code, but will you need to get the data again now? Seems like you can just pull out everything you need from the `prInfo` object